google-assistant-sdk/googlesamples/assistant/grpc/pushtotalk.py

```diff
@@ -317,21 +317,18 @@ def main(api_endpoint, credentials, project_id,
     logging.info('Connecting to %s', api_endpoint)
 
     # Configure audio source and sink.
-    audio_device = None
     if input_audio_file:
         audio_source = audio_helpers.WaveSource(
             open(input_audio_file, 'rb'),
             sample_rate=audio_sample_rate,
             sample_width=audio_sample_width
         )
     else:
-        audio_source = audio_device = (
-            audio_device or audio_helpers.SoundDeviceStream(
+        audio_source = audio_helpers.SoundDeviceStream(
```

Comment:
if we don't reuse the same stream for input/output, I think we might be able to get rid of some of the locking happening in conversationstream.

Reply:
Maybe reusing the stream is possible, I didn't understand how it could be reused, but now it is more clear for me :) Sorry, I am Python beginner :)